### PR TITLE
Monitoring Wazuh Statistics and metrics

### PR DESCRIPTION
**Pull request by Rebits**
# Description

This PR includes a script for monitoring statistics and metrics of wazuh analysisd, remoted and wazuhdb

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                             "time_execution", "time_global", "time_wazuhdb"])
+        elif process == "wazuh-remoted":
+            writer.writerow(["timestamp", "name", "bytes_received", "bytes_sent", "keys_reload_count",
+                             "received_keepalive",
+                             "received_request", "received_shutdown", "received_startup", "received_discarded",
+                             "received_event", "sent_ack", "sent_ar", "sent_discarded", "sent_request", "sent_sca",
+                             "sent_shared", "queue_size", "queue_usage"])
+        elif process == "wazuh-analysisd":
+            writer.writerow(["timestamp", "name", "bytes_received", "processed_events", "processed_events_received",
+                             "decoded_agent", "syscheck", "dropped_agent", "dropped_syscheck",
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/scripts/monitoring.py:72]: `syscheck` is too generic

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        writer = csv.writer(file)
+        if row:
+            writer.writerow(row)
+        else:
+            writer.writerow([timestamp, name, "No data"])
+
+
+def get_daemons_stats():
+    host = "localhost"
+    endpoint = "/manager/daemons/stats?daemons_list=wazuh-db,wazuh-analysisd,wazuh-remoted"
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/scripts/monitoring.py:168]: Hardcoded endpoint

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    logger.info(f"Starting metrics monitoring for processes: {options.process_list}")
+    metrics_monitoring_process = subprocess.Popen(f"wazuh-metrics -p {' '.join(options.process_list)} "
+                                                  f"--store {monitoring_evidences_directory}",
+                                                  shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    logger.info(f"Starting metrics monitoring for processes: {options.process_list}")
+    metrics_monitoring_pid = metrics_monitoring_process.pid
+    statistics_monitoring_thread = threading.Thread(target=collect_data,
+                                                    args=(options, monitoring_evidences_directory,))
+    statistics_monitoring_thread.start()
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/scripts/monitoring.py:248]: Same logger for metrics and statistics

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py (hunk 1 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/scripts/monitoring.py]: Done c24d1469c879ed2f2cc4947fda97718c491ad5af

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py (hunk 2 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/scripts/monitoring.py]: Done c24d1469c879ed2f2cc4947fda97718c491ad5af

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py (hunk 3 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/scripts/monitoring.py]: Done ab997c4a387846b4818a663df82a282b6378e7bb

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+            writer.writerow(["timestamp", "name", "bytes_received", "bytes_sent", "keys_reload_count",
+                             "received_keepalive",
+                             "received_request", "received_shutdown", "received_startup", "received_discarded",
+                             "received_event", "sent_ack", "sent_ar", "sent_discarded", "sent_request", "sent_sca",
+                             "sent_shared", "queue_size", "queue_usage"])
+        elif process == "wazuh-analysisd":
+            writer.writerow(["timestamp", "name", "bytes_received", "processed_events", "processed_events_received",
+                             "decoded_agent", "Decoded Events Syscheck", "dropped_agent", "dropped_syscheck",
+                             "writte_breakdown_alerts", "queue_size_alerts", "queue_usage_alerts",
+                             "queue_syscheck_size", "queue_syscheck_usage"])
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/scripts/monitoring.py:75]:
```suggestion
            writer.writerow(["timestamp", "name", "bytes_received", "processed_events", "processed_events_received",
                             "decoded_agent", "decoded_syscheck_events", "dropped_agent", "dropped_syscheck",
                             "writte_breakdown_alerts", "queue_size_alerts", "queue_usage_alerts",
                             "queue_syscheck_size", "queue_syscheck_usage"])
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/scripts/monitoring.py (hunk 4 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/scripts/monitoring.py]: Done in 6a5a9a0b08b16df2ba3fb3ad606b875be998b05d

**Review by juliamagan — APPROVED** [review]

**Pull request merged**